traitsui/testing/tester/qt4/implementation/enum_editor.py

```diff
@@ -0,0 +1,180 @@
+    index counts over the grid in the correct direction.  
+    The grid is always populated in row major order. The row_major trait of a
+    Radio Enum Editor simply changes what elements are assigned to each entry
+    in the grid, so that when displayed, they appear in column major order.
+    Qlayouts are indexed in the order they are populated, so to access
+    the correct element we may need to convert a column-major based index
+    into a row-major one.
+
+    Parameters
+    ----------
+    layout : QGridLayout
+        The layout of interest
+    index : int
+        the index of interest
+    row_major : bool
+        whether or not the grid entries are organized in row major order
+    """
+    if row_major:
+        return index
+    else:
+        n = layout.count()
+        num_cols = layout.columnCount()
+        num_rows = layout.rowCount()
+        # the last entries of the last row can be empty. 
+        num_empty_entries_last_row = num_cols * num_rows - n
+
+        # count in column major order
+
+        # if the index of interest extends into the part of the grid where
+        # the columns have a missing entry in the last row 
+        if index > num_rows * (num_cols - num_empty_entries_last_row):
+            # break the grid up into 2 grids.  One of size 
+            # num_rows x (num_cols - num_empty_entries_last_row).  The other
+            # of size (num_rows-1) x num_empty_entries_last_row
+            num_entries_grid1 = index - num_rows * (num_cols - num_empty_entries_last_row)
+            # find i, j coordinates of the index in grid2 if we counted in
+            # column major order
+            new_index = index - num_entries_grid1
+            i = new_index % (num_rows - 1)
+            j = new_index // (num_empty_entries_last_row)
+            # convert that back to an index found from row major order
+            return num_entries_grid1 + (i * num_empty_entries_last_row + j)
+        else:
+            # find i,j coordinates of index if we counted in column major order
+            i = index % num_rows
+            j = index // num_rows
+        # convert that back to an index found from row major order
+        return i * num_cols + j
+
+
+
+class _IndexedSimpleEditor(_BaseSourceWithLocation):
+    """ Wrapper class for Simple EnumEditor and Index.
+    """
+    source_class = SimpleEditor
+    locator_class = locator.Index
+    handlers = [
+        (command.MouseClick, (lambda wrapper, _: helpers.mouse_click_combobox(
+            combobox=wrapper.target.source.control,
+            index=wrapper.target.location.index,
+            delay=wrapper.delay))),
+    ]
+
+
+def radio_displayed_selected_text_handler(wrapper, interaction):
+    """ Handler function used to query DisplayedText for EnumRadioEditor.
+
+    Parameters
+    ----------
+    wrapper : UIWrapper
+        The UIWrapper containing that object with text to be displayed.
+    interaction : query.DisplayedText
```

Comment:
Reminder: This `DisplayedText` will need updating, also in the description of the function.